src/main/java/ch/njol/skript/util/Timespan.java

```diff
@@ -179,7 +196,50 @@ else if (length == 3 && !hasMs || length == 4) // HH:MM:SS[.ms]
 
 		return new Timespan(t);
 	}
-	
+
+
+	/**
+	 * Parses a shortened timespan (e.g., "5m", "3d", "1h") based on the provided input string and parse context.
+	 * Returns null if the input does not represent a valid shortened timespan in the context of a command argument.
+	 *
+	 * @param input   The input string representing the shortened timespan.
+	 * @param context The parse context to determine if parsing is allowed.
+	 * @return A Timespan object representing the parsed timespan, or null if parsing fails or not allowed in context.
+	 */
+	@Nullable
+	public static Timespan parseShortenedTimespan(String input, ParseContext context) {
+		if (context != ParseContext.COMMAND) {
+			return null;
+		}
+
+		long totalMillis = 0;
+		Matcher matcher = TIMESPAN_SHORTENED_PATTERN.matcher(input);
+
+		while (matcher.find()) {
+			int value = Integer.parseInt(matcher.group(1));
+			char unit = matcher.group(2).charAt(0);
```

Comment:
The regex pattern will pick up a match within a larger string (e.g. `abc(1d)ef`) which is good if you want to extract multiple matches (like `1d 2h`) but will also pick up things it probably shouldn't (e.g. `(10m)onths`).
<img width="165" alt="image" src="https://github.com/SkriptLang/Skript/assets/14147477/eebd9a3b-e1d4-4b6e-b686-2941a4fba9fe">
You can either limit this to just one match of the entire input (using the ^ and $ start and end tokens) or you can use some logic to make sure it's not the start of a larger text (e.g. `(?!\\w)`).

Reply:
Sovde told me to go a different route, and allow shorter versions in the default.lang, and then let the parser handle it. Which i'll try to fix up and finish today, as i didnt have much time yesterday.